lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,802 @@
+values in HTML and XML documents without resorting to typically verbose and
+inflexible DOM traversal methods.
+
+=head2 Redirects
+
+The L<Mojo::UserAgent> object in L<Test::Mojo> can handle HTTP redirections
+internally to whatever level you need. Let's say we have a web service that
+redirects C</1> to C</2>, C</2> redirects to C</3>, C</3> redirects to C</4>,
+and C</4> redirects to C</5>:
+
+  GET /1
+
+returns:
+
+  302 Found
+  Location: /2
+
+and:
+
+  GET /2
+
+returns:
+
+  302 Found
+  Location: /3
+
+and so forth, up to C</5>:
+
+  GET /5
+
+which returns the data we wanted:
+
+  200 OK
+
+  {"message":"this is five"}
+
+We can tell the user agent in L<Test::Mojo> how to deal with redirects. Each
+test is making a request to C<GET /1>, but we vary the number of redirects the
+user agent should follow with each test:
+
+  my $t = Test::Mojo->new;
+
+  $t->get_ok('/1')
+    ->header_is(location => '/2');
+
+  $t->ua->max_redirects(1);
+  $t->get_ok('/1')
+    ->header_is(location => '/3');
+
+  $t->ua->max_redirects(2);
+  $t->get_ok('/1')
+    ->header_is(location => '/4');
+
+  # look at the previous hop
+  is $t->tx->previous->res->headers->location, '/3', 'previous redirect';
+
+  $t->ua->max_redirects(3);
+  $t->get_ok('/1')
+    ->header_is(location => '/5');
+
+  $t->ua->max_redirects(4);
+  $t->get_ok('/1')
+    ->json_is('/message' => 'this is five');
+
+When we set C<max_redirects>, it stays set for the life of the test object
+until we change it.
+
+L<Test::Mojo>'s handling of HTTP redirects eliminates the need for making
+many, sometimes an unknown number, of redirections to keep testing precise and
+easy to follow (ahem).
+
+=head2 Cookies and session management
+
+We can use L<Test::Mojo> to test applications that keep session state in
+cookies. By default, the L<Mojo::UserAgent> object in L<Test::Mojo> will
+manage session for us by saving and sending cookies automatically, just like
+common web browsers:
+
+  use Mojo::Base -strict;
+
+  use Test::Mojo;
+  use Test::More;
+
+  my $t = Test::Mojo->new('MyApp');
+
+  # No authorization cookie
+  $t->get_ok('/')
+    ->status_is(401)
+    ->content_is('Please log in');
+
+  # Application sets an authorization cookie
+  $t->post_ok('/login', { password => 'let me in' })
+    ->status_is(200)
+    ->content_is('You are logged in');
+
+  # Sends the cookie from the previous transaction
+  $t->get_ok('/')
+    ->status_is(200)
+    ->content_like(qr/You logged in at \d+/);
+
+  # Clear the cookies
+  $t->reset_session;
+
+  # No authorization cookie again
+  $t->get_ok('/')
+    ->status_is(401)
+    ->content_is('Please log in');
+
+We can also inspect cookies in responses for special values through the
+transaction's response (L<Mojo::Message::Response>) object:
+
+  $t->get_ok('/');
+
+  my ($cookie) = grep { $_->name eq 'smarty' } @{$t->tx->res->cookies};
+  like($cookie && $cookie->to_string, qr(smarty=pants), 'cookie found');
```

Comment:
Not linking this example very much, it's kinda ugly. Maybe something like `like $t->tx->res->cookie('smarty'), qr/smarty=pants/, 'cookie found';`?